src/runtime/ranger/ranger_resource_policy.h

```diff
@@ -68,8 +100,36 @@ struct ranger_resource_policy
     std::unordered_set<std::string> table_names;
     acl_policies policies;
 
-    DEFINE_JSON_SERIALIZATION(name, database_names, table_names, policies)
+    DEFINE_JSON_SERIALIZATION(name, database_names, table_names, policies);
 };
 
+// A policy data structure definition of the DATABASE_TABLE resource, which will be set in
+// 'app_envs'
+struct matched_database_table_policy
+{
+    std::string matched_database_name;
+    std::string matched_table_name;
+    acl_policies policies;
+
+    DEFINE_JSON_SERIALIZATION(matched_database_name, matched_table_name, policies);
+};
+
+// Returns true if 'policies' allows "user_name" to access "database_name" via "rpc_code".
+// 'is_need_match_database' being true means that the 'policies' needs to be matched to the database
+// first, false means not
+bool check_ranger_resource_policy_allowed(const std::vector<ranger_resource_policy> &policies,
+                                          const access_type &ac_type,
+                                          const std::string &user_name,
+                                          bool is_need_match_database,
```

Comment:
As I've ever mentioned, it would be better to use enum rather than boolean.

Further more, add some comments to explain when needs to match database, and when doesn't need.